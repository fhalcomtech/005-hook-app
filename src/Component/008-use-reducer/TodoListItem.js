import React from 'react';

const TodoListItem = ({handlerClickDone, handlerClickDelete, todo, index}) => {
    const {id, desc, done} = todo;
    return (
        <li className="list-group-item d-flex justify-content-around" key={id}>
            <p
                onClick={(e)=>{handlerClickDone({id, desc, done:!done})}}
                className={ done && 'text-decoration-line-through text-success' }
            >
                {index + 1}. {desc}
            </p>
            <button className="btn btn-danger" onClick={(e)=>{
                handlerClickDelete({id,desc,done})
            }}>delete</button>
        </li>
    );
};

export default TodoListItem;
