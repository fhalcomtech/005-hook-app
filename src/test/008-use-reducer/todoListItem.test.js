import React from "react"
import { shallow } from "enzyme"
import TodoListItem from "./../../Component/008-use-reducer/TodoListItem";
import { initialTodoState } from "../fixtures";


describe("Testing todoListItem",()=>{
    const handlerClickDelete = jest.fn();
    const handlerClickDone = jest.fn();
    
    const wrapper = shallow(
    <TodoListItem
        todo={initialTodoState[0]}
        index={0}
        handlerClickDelete={handlerClickDelete}
        handlerClickDone={handlerClickDone}/>
        );

    test("Testing todoListItem create snapshot",()=>{
        expect(wrapper).toMatchSnapshot()
    })

    test("Testing todoListItem toggle done",()=>{
        wrapper.find("li > p").simulate("click");
        expect(handlerClickDone).toBeCalledWith({...initialTodoState[0],done: !initialTodoState[0].done});
    })

    test("Testing todoListItem must be correct text",()=>
    {
        const text = wrapper.find("li > p").text().trim();
        const textResult = `${initialTodoState[0].id}. ${initialTodoState[0].desc}`;
        expect(text).toEqual(textResult);
    })

    test("Testing todoListItem to delete a todo",()=>
    {
        wrapper.find("li button").simulate("click");
        expect(handlerClickDelete).toBeCalledWith(initialTodoState[0]);
    })


    test("Testing todoListItem to must be text-decoration-line-through class",()=>
    {
        const wrapper2 = shallow(
            <TodoListItem
                todo={initialTodoState[0]}
                index={0}
                handlerClickDelete={handlerClickDelete}
                handlerClickDone={handlerClickDone}/>
                );
    })
})