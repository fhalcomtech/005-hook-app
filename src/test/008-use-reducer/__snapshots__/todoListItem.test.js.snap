// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing todoListItem Testing todoListItem create snapshot 1`] = `
<li
  className="list-group-item d-flex justify-content-around"
  key="1"
>
  <p
    className={false}
    onClick={[Function]}
  >
    1
    . 
    Learn React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    delete
  </button>
</li>
`;
